dpctl/tensor/_reduction.py

```diff
@@ -369,6 +443,8 @@ def _comparison_over_axis(x, axis, keepdims, _reduction_fn):
         axis = tuple(range(nd))
     if not isinstance(axis, (tuple, list)):
         axis = (axis,)
+    if any([x.shape[i] == 0 for i in axis]):
```

Comment:
I would suggest making the same type of change as suggest above, which avoids call to `permute_dims` is `axis=None`